core/sail/nativerdf/src/test/java/org/eclipse/rdf4j/sail/nativerdf/NativeStoreTxnTest.java

```diff
@@ -104,4 +109,35 @@ protected void removeStmt(Statement stmt) {
 			conn.commit();
 		}
 	}
+
+	@Test
+	public void testOldTxnStatusFile() throws Exception {
+
+		try (RepositoryConnection con = repo.getConnection()) {
+			con.begin();
+			assertFalse(con.hasStatement(RDF.TYPE, RDF.TYPE, RDFS.RESOURCE, false));
+			con.add(RDF.TYPE, RDF.TYPE, RDFS.RESOURCE);
+			con.commit();
+		}
+		repo.shutDown();
+
+		File txnStatusFile = new File(repo.getDataDir().getAbsolutePath() + "/txn-status");
+
+		// write old format of txn-status
+		NioFile nioFile = new NioFile(txnStatusFile);
+		byte[] bytes = "COMMITTING".getBytes(US_ASCII);
+		nioFile.truncate(bytes.length);
+		nioFile.writeBytes(bytes, 0);
+
+		repo.init();
+
+		try (RepositoryConnection con = repo.getConnection()) {
+
+			con.begin();
+			assertTrue(con.hasStatement(RDF.TYPE, RDF.TYPE, RDFS.RESOURCE, false));
+			con.commit();
```

Comment:
Do you want to put a further assertion here that after replaying the txn and getting everything in shape, the transaction file now uses the new format?

Reply:
I found the TripleStoreRecoveryTest and based my test on that one, with the exception that I hardcoded the old "COMMITTING" string.